pageserver/src/tenant/timeline.rs

```diff
@@ -4175,7 +4183,9 @@ impl Timeline {
                 layer_names_to_delete.push(doomed_layer.filename());
                 result.layers_removed += 1;
             }
-            let apply = guard.finish_gc_timeline(layer_removal_cs, gc_layers, &self.metrics)?;
+            guard
```

Comment:
`ApplyGcGuard` from [the previous PR](https://github.com/neondatabase/neon/pull/4618) is now removed because at this point we are holding the write lock and therefore it does not matter whether to update the layer map at this point or after scheduling uploads.